libfreerdp/crypto/tls.c

```diff
@@ -872,6 +872,18 @@ int tls_connect(rdpTls* tls, BIO* underlying)
 	return tls_do_handshake(tls, TRUE);
 }
 
+BOOL tls_prep(rdpTls* tls,BIO* underlying, int options, BOOL clientMode)
```

Comment:
This function shouldn't exist, you should do the method selection stuff in `tls_connect()`. If we have this `EnforceTLSv1_2` option it should probably also apply for server-side connections

Reply:
@hardening as far as I see it this is just a `hack` to fix broken behavior of windows servers?

Reply:
@hardening  
> If we have this EnforceTLSv1_2 option it should probably also apply for server-side connections
Not sure if you are suggesting a code change here in FreeRDP but the target server I'm trying to connect to already has TLS1.2 enabled. The issue is very specific to the win 2019 server and only occurs exclusively when both TLS1.2 and TLS1.3 are enabled - almost as if this machine is misleading FreeRDP during the handshake and then resulting in a `RST` response. 

FWIW, both win2019 and win2022 server machines can't be accessed even from the official Microsoft RDP client when TLS1.3 is the only TLS version enabled - maybe RDP can't solely function on TLS1.3 ? idk 🤷‍♂️ at least we are on parity with the official RDP client in this regard :)